vault/resource_token.go

```diff
@@ -89,6 +89,12 @@ func tokenResource() *schema.Resource {
 				Default:     "token",
 				Description: "The display name of the token.",
 			},
+			"metadata": {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				ForceNew:    true,
```

Comment:
Should we use `ForceNew` here? It seems to me like even if metadata changes, it still could essentially be the same object, we wouldn't need a brand new one.

Reply:
Hm, yes, good point, I do see that the `meta` currently can't be changed by any of the token endpoints after it's been created.

However, that's not a guarantee going forward. In the future there may be a valid use case for the metadata to change without the token itself changing, for example, if we wanted to enrich tokens with additional metadata for audit logs. In theory, the metadata _could_ change without require a brand new token, it's not part of the fields contract or documentation that it _must_ always be static on a token.

Reply:
I did a quick test setting `ForceNew: false` to see how Terraform would behave. I created a token with some metadata then added/removed some of the keys. On the first apply after editing the Terraform hcl code, it said it updated the token. But on every single apply after that, Terraform said it's making the same change again and again. I don't think that's desirable behavior; it doesn't line up with my understanding of Terraform. The future use case you mentioned makes sense but I think the provider should be tailored to the constraints Vault has today.

I noticed some other properties both in this resource and others in the Vault provider that exhibit this behavior - they're written in a way that lets Terraform say a change was made and nothing happened - maybe I'm missing something?

Reply:
Hi! Regarding `ForceNew`, its functionality is documented [here](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#forcenew) and I'm afraid I'm not going to be comfortable merging this until it's not set to false, for the reasons stated earlier. 

Having a permanent diff is a separate issue and can be addressed through the `SchemaDiffSuppressFunc` function.

Thanks again!

Reply:
Okay. I'm not comfortable with submitting code with behavior that I view as a bug. Open to the idea, but I don't understand why it's not a bug. So apologies, I must decline. Thanks for taking a look!